static/app/views/alerts/workflowEngineRedirects.tsx

```diff
@@ -70,7 +110,93 @@ function WorkflowEngineRedirectWithData({
   return children;
 }
 
-// Data-dependent redirects
+/**
+ * Base component for workflow engine redirects that require fetching
+ * workflow data from a metric alert rule before redirecting.
+ */
+function WorkflowEngineRedirectWithAlertRuleData({
```

Comment:
can we make this a hook (or small HoCish wrapper for class components) and wrap the areas we want to redirect instead. Adding these to the route tree doesn't totally make sense if they aren't views